crates/toolchain/instructions/src/program.rs

```diff
@@ -58,6 +58,27 @@ impl<F: Field> Program<F> {
         }
     }
 
+    pub fn new_without_debug_infos_with_option(
```

Comment:
`new_sparse`?

I feel like we can drop `without_debug_infos` (also from the other constructor), since that's more like the default, and `with_debug_infos` is extra descriptor